### DISCUSSION TITLE
Play and call in a single actor

### DISCUSSION THREAD

**Discussion by pboling** [sunny/actor#123] [Q&A]
Does it make sense?  Can it be done?  Is anyone doing this?  It seems like this is what I would want to use, because I want to dynamically declare one of the plays based on inputs like;

```
# not working pseudo code
class Streamer < Actor
  input :provider

  play "Streaming::Provider::#{provider.classify}".constantize
end
```

The input is available from `call`, of course, but there are zero examples anywhere I can find of an Actor using both the `call` and the `play` patterns.

**Comment by sunny** [comment]
Interesting use case! In this simple example I would do the following:

```rb
class Streamer < Actor
  input :provider

  def call
    "Streaming::Provider::#{provider.classify}".constantize.call
  end
end
```

If you want to be able to use `play` and pass the inputs and the result from one actor to the other, however, you can do something like this instead, by reusing the `result`:

```rb
class Streamer < Actor
  input :provider

  play :call_provider

  private

  def call_provider
    "Streaming::Provider::#{provider.classify}".constantize.call(result)
  end
end
```
